lambdas/services/generate_document_manifest_zip_service.py

```diff
@@ -0,0 +1,120 @@
+        except ClientError as e:
+            self.update_failed_status()
+            msg = f"{file_key} may reference missing file in s3 bucket: {file_bucket}"
+            logger.error(
+                f"{LambdaError.ZipServiceClientError.to_str()} {msg + str(e)}",
+                {"Result": "Failed to create document manifest"},
+            )
+            raise DocumentManifestServiceException(
+                status_code=500, error=LambdaError.ZipServiceClientError
+            )
+
+    def get_file_bucket_and_key(self, file_location):
+        try:
+            file_bucket, file_key = file_location.replace("s3://", "").split("/", 1)
+            return file_bucket, file_key
+        except ValueError:
+            self.update_failed_status()
+            raise InvalidDocumentReferenceException(
+                "Failed to parse bucket from file location string"
+            )
+
+    def upload_zip_file(self):
+        logger.info("Uploading zip file to s3")
+        self.zip_trace_object.zip_file_location = (
+            f"s3://{self.zip_output_bucket}/{self.zip_file_name}"
+        )
+
+        try:
+            self.s3_service.upload_file(
+                file_name=self.zip_file_path,
+                s3_bucket_name=self.zip_output_bucket,
+                file_key=f"{self.zip_file_name}",
+            )
+        except ClientError as e:
+            self.update_failed_status()
+            logger.error(e, {"Result": "Failed to create document manifest"})
+            raise DocumentManifestServiceException(
+                status_code=500, error=LambdaError.ZipServiceClientError
+            )
+
+        self.zip_trace_object.job_status = ZipTraceStatus.COMPLETED
+
+    def zip_files(self):
+        logger.info("Creating zip from files")
+        with zipfile.ZipFile(self.zip_file_path, "w", zipfile.ZIP_DEFLATED) as zipf:
+            for root, _, files in os.walk(self.temp_downloads_dir):
+                for file in files:
+                    file_path = os.path.join(root, file)
+                    arc_name = os.path.relpath(file_path, self.temp_downloads_dir)
+                    zipf.write(file_path, arc_name)
+
+    def update_dynamo_with_fields(self, fields: set):
+        logger.info("Writing zip trace to db")
+        self.dynamo_service.update_item(
+            self.zip_trace_table,
+            self.zip_trace_object.id,
+            self.zip_trace_object.model_dump(by_alias=True, include=fields),
+        )
+
+    def update_processing_status(self):
+        self.zip_trace_object.job_status = ZipTraceStatus.PROCESSING
+        self.update_dynamo_with_fields({"job_status"})
+
+    def update_failed_status(self):
+        self.zip_trace_object.job_status = ZipTraceStatus.FAILED
+        self.update_dynamo_with_fields({"job_status"})
+
+    def remove_temp_files(self):
+        # Removes the parent of each removed directory until the parent does not exist or the parent is not empty
```

Comment:
Is this comment necessary? I would maybe rename this function to "cleanup_temp_files" or something similar to describe what is happening here